### PR TITLE
test: migrate progress bar, selected option and radio button tests

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Task [link](https://textkernel.atlassian.net/browse/ONEUI-394)

### CODE REVIEW

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { SelectedOption } from '../SelectedOption';
+
+describe('<SelectedOption>', () => {
+    const onDeleteMock = jest.fn();
+
+    let view: RenderResult;
+
+    afterEach(() => {
```
Comment (eszthoff) [inline src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:12]: I don't think we need this anymore. It was an older version of jest that required manual reset.

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        jest.clearAllMocks();
+    });
+
+    it('should render nothing if no children are provided', () => {
+        view = render(<SelectedOption onDelete={onDeleteMock} />);
+
+        expect(view.container).toMatchSnapshot();
+        expect(view.container.children).toHaveLength(0);
```
Comment (eszthoff) [inline src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:20]: s/a

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/ProgressBar/__tests__/ProgressBar.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('progressbar')).toBeInTheDocument();
+    });
+
+    it('should render a progress bar with label', () => {
+        view.rerender(<ProgressBar percentage={25}>Loading...</ProgressBar>);
+
+        const { children } = view.container;
+        // Check that label is shown
+        expect(children[0].textContent).toBe('Loading...');
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/ProgressBar/__tests__/ProgressBar.spec.tsx:21]: :+1:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 4]
```diff
@@ -19,29 +21,48 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
-        expect(toJson(wrapper)).toMatchSnapshot();
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getAllByRole('radio')).toHaveLength(2);
     });
+
     it('should pass name prop to children', () => {
-        wrapper
-            .find('RadioButton')
-            .forEach((option) => expect(option.prop('name')).toEqual('group_name'));
+        const namePropValue = 'group_name';
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:29]: :+1:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 5]
```diff
@@ -19,29 +21,48 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
+                <RadioButton onChange={onChangeGlobalMock} id="o1">
+                    Option 1
+                </RadioButton>
+                <RadioButton id="o2">Option 2</RadioButton>
+            </RadioButtonGroup>
+        );
+
+        const buttons = screen.getAllByRole('radio');
+
+        expect(buttons).toHaveLength(4);
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:62]: :+1:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 6]
```diff
@@ -19,29 +21,48 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
+        expect(onChangeChildMock).toHaveBeenCalled();
     });
-    it('should pass onChange prop to children', () => {
+
+    it('should pass onChange prop to children', async () => {
+        const user = userEvent.setup();
         const onChangeGlobalMock = jest.fn();
-        wrapper.setProps({ onChange: onChangeGlobalMock });
-        wrapper.update();
+        view = render(
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:51]: :+1:

File: src/components/ProgressBar/__tests__/ProgressBar.spec.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/ProgressBar/__tests__/ProgressBar.spec.tsx:21]: when i use         expect(screen.getByText('Loading...')).toBeInTheDocument();
i get error:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 7]
```diff
@@ -43,26 +44,26 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
         const buttons = screen.getAllByRole('radio');
 
-        expect(buttons).toHaveLength(4);
+        expect(buttons).toHaveLength(2);
 
-        await user.click(buttons[2]);
+        await user.click(buttons[0]);
 
         expect(onChangeGlobalMock).toHaveBeenCalled();
+        expect(buttons[1]).not.toHaveAttribute('onChange');
```
Comment (eszthoff) [inline src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:67]: hhmm. This is not what should happen. Maybe this check is not a good one, if it passes both here and in the test above. 

Here, you are trying to see that the global handler is passed to the children and is triggered when you click on the radio button.

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx [hunk 8]
```diff
@@ -9,28 +9,25 @@ describe('<SelectedOption>', () => {
...
-    afterEach(() => {
-        jest.clearAllMocks();
-    });
-
     it('should render nothing if no children are provided', () => {
         view = render(<SelectedOption onDelete={onDeleteMock} />);
 
         expect(view.container).toMatchSnapshot();
-        expect(view.container.children).toHaveLength(0);
+        expect(view.container).toBeInTheDocument();
```
Comment (eszthoff) [inline src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:16]:
```suggestion
        expect(view.container).toBeEmptyDOMSomething();
```

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx [hunk 9]
```diff
@@ -9,28 +9,25 @@ describe('<SelectedOption>', () => {
...
     });
 
     it('should render children if it is 0', () => {
         view = render(<SelectedOption onDelete={onDeleteMock}>{0}</SelectedOption>);
 
-        expect(view.container.children).toHaveLength(1);
+        expect(view.container).toBeInTheDocument();
```
Comment (eszthoff) [inline src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:22]: not sure what is it that you are testing with this line

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 10]
```diff
@@ -1,14 +1,18 @@
...
-    let wrapper;
+    let view: RenderResult;
+
     const onChangeChildMock = jest.fn();
+    const namePropValue = 'group_name';
 
     beforeEach(() => {
-        wrapper = mount(
+        view = render(
             <RadioButtonGroup name="group_name">
```
Comment (eszthoff) [inline src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:16]:
```suggestion
            <RadioButtonGroup name={namePropValue}>
```

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx (hunk 10 above)
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:16]: :+1:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx (hunk 7 above)
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:67]: Yes and check that it contains onChangeGlobalMock for onChange

Comment (eszthoff) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:67]: but the test say `not.toHaveAttribute('onChange')`.... 🤔

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx (hunk 9 above)
Comment (eszthoff) [reply src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:22]: How does this line tests that? Asking as it neither mentions `0` nor `text`

File: src/components/SelectedOption/__tests__/SelectedOption.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+        expect(view.container).toMatchSnapshot();
+        expect(view.container).toBeEmptyDOMElement();
+    });
+
+    it('should render children if it is 0', () => {
+        view = render(<SelectedOption onDelete={onDeleteMock}>{0}</SelectedOption>);
+
+        expect(view.container).toHaveTextContent('0');
+    });
```
Comment (eszthoff) [inline src/components/SelectedOption/__tests__/SelectedOption.spec.tsx:23]: but your idea of checking if 0 is there was good. What happened to it?
expect(view.container).toHaveTextContent('0');

**Review by eszthoff — CHANGES_REQUESTED** [review]

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 12]
```diff
@@ -19,29 +23,50 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
+        expect(buttons).toHaveLength(2);
+
+        await user.click(buttons[0]);
+
+        expect(onChangeChildMock).toHaveBeenCalled();
+        expect(buttons[1]).not.toHaveAttribute('onChange');
     });
-    it('should pass onChange prop to children', () => {
+
+    it('should onChange should called for all children', async () => {
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:50]: :+1:

File: src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx [hunk 13]
```diff
@@ -19,29 +23,50 @@ describe('<RadioButtonGroup> that renders a group of radio buttons', () => {
...
+    it('should not modify onChange prop of children if they are no defined on the group level', async () => {
+        const user = userEvent.setup();
+        const buttons = screen.getAllByRole('radio');
+
+        expect(buttons).toHaveLength(2);
+
+        await user.click(buttons[0]);
+
+        expect(onChangeChildMock).toHaveBeenCalled();
+        expect(buttons[1]).not.toHaveAttribute('onChange');
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/__tests__/RadioButtonGroup.spec.tsx:47]: :+1:

**Review by eszthoff — APPROVED** [review]

**Pull request merged**